src/m2n/M2N.cpp

```diff
@@ -69,6 +78,15 @@ void M2N::requestPrimaryRankConnection(
     PRECICE_DEBUG("Request primary connection");
     _interComm->requestConnection(acceptorName, requesterName, "PRIMARYCOM", 0, 1);
     _isPrimaryRankConnected = _interComm->isConnected();
+
+    //check that local and remote have the same precice version
+    std::string requesterVersion = PRECICE_VERSION;
+    auto        acceptorVersion  = requesterVersion;
+    _interComm->send(requesterVersion, 0);
+    _interComm->receive(acceptorVersion, 0);
```

Comment:
These two should be swapped as we are expecting send and receive to be blocking.